### PR TITLE
Add channel + company + workspace models

### PR DESCRIPTION
**Pull request by manhduc-nguyen**

### CODE REVIEW

File: lib/models/channel/channel.dart [hunk 1]
```diff
@@ -1 +1,55 @@
...
+    required this.membersCount,
+    required this.members,
+    required this.visibility,
+    required this.lastActivity,
+    required this.userLastAccess,
+    this.draft,
+    required this.permissions
+  });
+
+  factory Channel.fromJson(Map<String, dynamic> json) => _$ChannelFromJson(json);
```
Comment (bobs4462) [inline lib/models/channel/channel.dart:52]: We have a problem here, it's related to SQLite datatypes, and there's only four of them: TEXT, INT, REAL, BLOB. 
And List<T> and Map<K,V> is not one of them. So if we were to store those model to database will get errors. 
So I suggest to add extra logic to convert those types to string and back during the storage and retrieval from DB. 
You can refer to Message model for reference as usual.

Comment (bobs4462) [reply lib/models/channel/channel.dart:52]: https://github.com/linagora/Twake-Mobile/blob/V3/lib/models/message/message.dart

Comment (manhduc-nguyen) [reply lib/models/channel/channel.dart]: @bobs4462 done

**Comment by RockinPaul** [comment]
@manhduc-nguyen have you finished your updates of this PR to comply with the new models?

**Comment by manhduc-nguyen** [comment]
> @manhduc-nguyen have you finished your updates of this PR to comply with the new models?

Yes its done

**Review by bobs4462 — APPROVED** [review]

**Pull request merged**